feat(model): add iML1515 model, Memote CI, and update documentation

KennetLindquist opened the pull request:
### Main improvements in this PR:
-  **Added iML1515 model** from [BiGG Models](http://bigg.ucsd.edu/models/iML1515) (`feat(model): add iML1515.xml`)
       The BiGG database also provides `.json` and `.mat` model files for iML1515 which are currently excluded in this repository because their contents differ. The `.xml` file was chosen as a baseline because it is the most comprehensive, and any missing information can be added to it. Future releases will document these differences and provide guidance on their handling.
-  **Added Memote CI** (`ci(memote): add memote_test.yml`)
-  **Updated documentation** (`docs: update checklist status in standard-GEM.md`)

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model

KennetLindquist committed:
docs: update checklist status in .standard-GEM.md

Checklist items marked as complete:
- All GEMs that follow the `standard-GEM` must contain this file.
- Navigate to [standard-GEM](https://github.com/MetabolicAtlas/standard-GEM/) and click on the button `Use this template`
- Pick a repository name 
- Pick a repository description
- Add repository topic
- Git branches

KennetLindquist committed:
docs: update checklist status in .standard-GEM.md

Checklist items marked as complete:
- Commits

KennetLindquist committed:
feat(model): add iML1515.xml 

Add iML1515 model from BiGG (http://bigg.ucsd.edu/models/iML1515).

KennetLindquist committed:
ci(memote): add memote_test.py

Add Python script for Memote testing.

KennetLindquist committed:
ci(memote): add memote_test.yml

Add GitHub Action workflow for memote testing on PRs to main.

KennetLindquist committed:
chore: rename memote_text.py to memote_test.py

KennetLindquist committed:
docs: update readme

Update readme with title, citations, and keywords.

KennetLindquist committed:
feat(model): add iML1515 model, Memote CI, and update documentation

feat(model): add iML1515 model, Memote CI, and update documentation

KennetLindquist committed:
chore(ci): ensure memote_test workflow checks out PR branch before running tests

Ensure memote_test workflow checks out PR branch before running tests

KennetLindquist commented:
Memote run failed due to an incompatibility with pull-requests in the memote_test.yml file. It is being resolved in the latest PR #4 from Initial_commit to develop.

cdebuck commented on README.md:
```diff
@@ -16,36 +7,27 @@ If you find this template does not fit your needs, we would appreciate if you co
 #### Description
 
-{{ fill in a short description or the paper abstract }}
+This repository contains the latest version the E. coli GEM, a genome scale metabolic model of _Escherichia coli_. 
```
It might be good to consider already mentioning the specific E. coli strain at this point in the Readme? (It is already part of the key-words section, so it is also fine like this, but different E. coli strains can behave quite differently, so it might be good to be explicit.

KennetLindquist replied:
It makes sense to mention it as early as possible to avoid confusion. PR #4 includes this suggested change.

cdebuck commented on README.md:
```diff
@@ -16,36 +7,27 @@ If you find this template does not fit your needs, we would appreciate if you co
-**Taxonomy ID:** {{ [taxonomy:559292](https://identifiers.org/taxonomy:559292) }}  
-**Genome ID:** {{ [insdc.gca:GCA_000146045.2](https://identifiers.org/insdc.gca:GCA_000146045.2)  }}  
+**Taxonomic name:** _Escherichia coli_   
+**Taxonomy ID:** [taxonomy:562](https://identifiers.org/taxonomy:562)   
```
Here we could also consider making the name and taxonomy ID more specific to the strain: https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=511145&lvl=3&lin=f&keep=1&srchmode=1&unlock

KennetLindquist replied:
Taxonomy name and ID now changed to specify strain in PR #4 .

cdebuck commented on code/memote_text.py:
```diff
@@ -0,0 +1,17 @@
+
+
+# Save html report
+report_path = 'memote_report.html'
```
Do we want the memote_report to end up in one of the subfolders of the repo? Or just in the root folder?

KennetLindquist replied:
It is intended for it to end up in the root folder.

KennetLindquist committed:
docs: update Taxonomic name and ID for E. coli in README

Update Taxonomic name and ID for E. coli to specific strain MG1655 in README.

KennetLindquist committed:
chore(ci): ensure memote_test workflow checks out PR branch before running tests

### Main improvements in this PR:
- Ensure memote_test workflow checks out PR branch before running tests
- Update Taxonomic name and ID to specify strain in README

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model
- [x] Selected `devel` as a target branch

KennetLindquist committed:
Add Memote HTML report [skip ci]

mihai-sysbio reviewed (approved):
Great job @KennetLindquist ! 

One tiny detail I've noticed is that the commit message of the GH Action that runs _memote_ stands out when looked at together with the other nicely-written commit messages.
